Azure/Java intro workshop (#5)
pulumi/workshops: azure-intro/java/content/lab-01/README.md

# Creating a New Project

Infrastructure in Pulumi is organized into projects. Each project is a single program that, when run, declares the desired infrastructure for Pulumi to manage.

## Step 1 &mdash; Create a Directory

Each Pulumi project lives in its own directory. From the root of this repository, create a new directory and change into it:

```bash
mkdir my-first-azure-app && cd my-first-azure-app
```

## Step 2 &mdash; Initialize Your Project

A Pulumi project is just a directory with some files in it. It's possible for you to create a new one by hand. The `pulumi new` command, however, automates the process:

```bash
pulumi new java -y
```

> This will print output similar to the following with a bit more status as it goes:

Author: Fixed.

```bash
Created project 'my-first-azure-app'

Please enter your desired stack name.
To create a stack in an organization, use the format <org-name>/<stack-name> (e.g. `acmecorp/dev`).
Created stack 'dev'

(some build output)

Your new project is ready to go!
```

This command has created all the files we need, initialized a new stack named `dev` (an instance of our project), and installed the needed package dependencies from Maven.

## Step 3 &mdash; Inspect Your New Project

Our project is comprised of multiple files:

* **`src/main/java/myproject/App.java`**: your program's main entry point
* **`pom.xml`**: your project's Maven dependency information
* **`Pulumi.yaml`**: your project's metadata, containing its name and language
* **`target/`**: a directory containing your project's built artifacts

Run `cat src/main/java/myproject/App.java` to see the contents of your project's empty program:

```java
package myproject;

import com.pulumi.Pulumi;
import com.pulumi.core.Output;

public class App {
    public static void main(String[] args) {
        Pulumi.run(ctx -> {
            ctx.export("exampleOutput", Output.of("example"));
        });
    }
}
```

> We can remove the following import from our `App.java` as it is not used in this workshop:

Author: ```suggestion
We can remove the following import from our `App.java` for now:
```
